boot/makebootable.go

```diff
@@ -167,20 +167,34 @@ func makeBootable20(model *asserts.Model, rootdir string, bootWith *BootableSet)
 		return fmt.Errorf("internal error: cannot find bootloader: %v", err)
 	}
 
-	// on e.g. ARM we need to extract the kernel assets on the recovery system
-	// as well
+	// on e.g. ARM we need to extract the kernel assets on the recovery
+	// system as well, but the bootloader does not load any environment from
+	// the recovery system
 	ekrbl, ok := bl.(bootloader.ExtractedRecoveryKernelImageBootloader)
 	if ok {
 		kernelf, err := snap.Open(bootWith.KernelPath)
 		if err != nil {
 			return err
 		}
 
-		return ekrbl.ExtractRecoveryKernelAssets(
+		err = ekrbl.ExtractRecoveryKernelAssets(
 			bootWith.RecoverySystemDir,
 			bootWith.Kernel,
 			kernelf,
 		)
+		if err != nil {
+			return fmt.Errorf("cannot extract recovery system kernel assets: %v", err)
+		}
+
+		// record which recovery system is to be used
+		// TODO:UC20: should we do this for all bootloaders?
```

Comment:
I think we should eventually

Reply:
we definitely must,  we are avoiding this because the recovery grub cfg is still essentially not what we want/it's cheating